cleaneval.py

```diff
@@ -0,0 +1,327 @@
+opt_summary = ('-s', '') in options
+opt_unlabelled = ('-u', '') in options
+opt_ascii = ('-a', '') in options
+
+
+
+
+def do_cleaneval(url, gold_html, cleaned_html):
+    '''
+    add docstring with test cases
+    '''
+    text_words = re_WS.split(cleaned_html)
+    gold_words = re_WS.split(gold_html)
+    alignment = SequenceMatcher(None, text_words, gold_words)
+    diff = make_diff(alignment, text_words, gold_words)
+    eval_list = evaluate(diff)
+    return eval_list
+
+def clean_cleaneval(raw_html):
+    cleaned_html = normalize(raw_html, opt_ascii, opt_unlabelled)
+    return cleaned_html
+
+def clean_metahtml(raw_html):
+    url = __get_url(raw_html)
+    res = metahtml.parse(raw_html, url, None, False, True)
+    cleaned_html = res['content']['best']['value']['text']
+    return cleaned_html
+
+def cleaneval_from_files(gold_path, raw_path, list_of_libraries = [ lambda x: x ], eval_method=do_cleaneval):
+    '''
+    gold_path is the path to the directory containing the gold standard files
+    raw_path is the path to the directory containing the raw files
+    '''
+    n_processed = 0
+    raw_files = os.listdir(raw_path)
+    n_files = len(raw_files)
+    eval_list = {}
+    sum = {}
+    ss = {}
+    for library in list_of_libraries:
+        sum[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ss[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    for filename in raw_files:
+        raw_file = raw_path + '/' + filename
+        gold_standard_file = gold_path + '/' + filename
+        if not os.access(gold_standard_file, os.R_OK):
+            continue  # original and cleaned files don't always match in CleanEval gold standard
+        raw_html = slurp_file(raw_file)
+        gold_html = normalize(slurp_file(gold_standard_file), opt_ascii, opt_unlabelled)
+        url = __get_url(raw_html)
+        for library in list_of_libraries:
+            cleaned_html = library(raw_html)
+            eval_list[library.__name__] = eval_method(url, gold_html, cleaned_html)
+            for i, x in enumerate(eval_list[library.__name__]):
+                sum[library.__name__][i] += x
+                ss[library.__name__][i] += x ** 2
+        n_processed += 1
+    if n_processed < n_files:
+        print("Warning: %d files not found in gold standard (skipped)" % (n_files - n_processed), file=sys.stderr)
+    print("Library\tF\tP\tR\tF.tag\tP.tag\tR.tag\tTP\tFP\tFN\tTP.tag\tFP.tag\tFN.tag")
+    for library in list_of_libraries:
+        tp = sum[library.__name__][6]
+        fp = sum[library.__name__][7]
+        fn = sum[library.__name__][8]
+        tag_tp = sum[library.__name__][9]
+        tag_fp = sum[library.__name__][10]
+        tag_fn = sum[library.__name__][11]
+        result = __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn)
+        print(library.__name__ + "\t" + ("%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%d\t%d\t%d\t%d\t%d\t%d" % result))
+
+#    if not os.access(gold_file, os.R_OK):
```

Comment:
the master branch of this repo cannot contain chunks of commented out code like this